javaCarAPI/src/main/java/api/CarAPI.java

```diff
@@ -0,0 +1,261 @@
+    private static final int STEERING_ANGLE = 50;
+    private static final int IDLE_SPEED = 0;
+    private static final int QOS = 1;
+    private static final int IMAGE_WIDTH = 320;
+    private static final int IMAGE_HEIGHT = 240;
+
+    private String throttleControl;
+    private String steeringControl;
+    private String cameraFeed;
+    private String jsonCommand;
+    private String ultrasonicFront;
+    private String gyroscope;
+    private String infrared;
+    private String odometerSpeed;
+    private String odometerTotalDistance;
+    private String heartbeat;
+
+    private final MqttClient client;
+    private final String carName;
+    private final String clientId;
+
+    private long ping = 0;
+
+    private final List<Consumer<Image>> videoFeedListeners = new ArrayList<>();
+    private final List<Consumer<Double>> ultraSonicListeners = new ArrayList<>();
+    private final Map<Infrared,List<Consumer<Double>>> infraredListeners = new HashMap<>();
+    private final Map<Odometer,List<Consumer<Double>>> odometerSpeedListeners = new HashMap<>();
+    private final Map<Odometer,List<Consumer<Double>>> odometerTotalDistanceListeners = new HashMap<>();
+    private final List<Consumer<Double>> gyroscopeListeners = new ArrayList<>();
+
+    public CarAPI(String host, String carName) throws MqttException{
+        this(host, UUID.randomUUID().toString(),carName);
+    }
+
+    public CarAPI (String host,String clientId,String carName) throws MqttException{
+        this(host,1883,clientId,carName);
+    }
+
+    public CarAPI (String host,int port,String clientId,String carName) throws MqttException{
+        this(host,port,clientId,carName,null,null);
+    }
+
+    public CarAPI (String host,int port,String clientId,String carName,String user, char [] password) throws MqttException{
+        this.carName = carName;
+        this.clientId = clientId;
+
+        initMQTTVariables();
+
+        String hostURL = "tcp://"+host+":"+port;
+        client = new MqttClient(hostURL,clientId, new MqttDefaultFilePersistence(System.getProperty("java.io.tmpdir")));
+        MqttConnectOptions options = new MqttConnectOptions();
+        options.setAutomaticReconnect(true);
+        options.setCleanSession(true);
+        options.setConnectionTimeout(10);
+        if (user != null && password != null) {
+            options.setUserName(user);
+            options.setPassword(password);
+        }
+        client.connect(options);
+
+        subscribe();
+        startHeartbeatThread();
+
+    }
+
+    private void startHeartbeatThread() {
+        Thread thread = new Thread(() -> {
+            while (client.isConnected()) {
+                try {
+                    MqttMessage currentTime = new MqttMessage(Long.toString(System.currentTimeMillis()).getBytes(StandardCharsets.UTF_8));
+                    client.publish(heartbeat,currentTime);
+                    final int PING_REQUEST_INTERVAL = 150;
```

Comment:
This constant could be moved to the top of the class so that it's easy to see and change or reuse in the future.